components/main-chef-wrapper/main.go

```diff
@@ -0,0 +1,123 @@
+//
+// Copyright 2019 Chef Software, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package main
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+)
+
+func main() {
+	// No arguments provided, display usage
+	if len(os.Args) <= 1 {
+		usage()
+		os.Exit(0)
+	}
+	var (
+		subCommand = os.Args[1]
+		allArgs    = os.Args[2:]
+		cmd        *exec.Cmd
+	)
+
+	switch subCommand {
+	// 1) On the creation of new CLIs, for instance, a new CLI called chef-analyze that
+	// will provide some analyze capabilities, as a user, you can run the binary directly
+	// 'chef-analyze foo' or through our main Chef CLI wrapper 'chef analyze foo'.
+	// Both ways would be running the same underlying binary.
+	case "analyze":
```

Comment:
I left this bit since we are going to merge this PR very soon: https://github.com/chef/chef-workstation/pull/553